### ISSUE TITLE
Symetric Key Attestation

### ISSUE THREAD

**Issue by duncanfinney** [Azure/iotedge#578]
Hi there! 

First of all, keep up the awesome work! My company is really excited to migrate our code to iot edge. 

I noticed that TPM and manual connection strings are the only officially supported methods for provisioning. 

What would happen if I use Symetric Key device provisioning as detailed in this document: https://docs.microsoft.com/en-us/azure/iot-dps/how-to-legacy-device-symm-key

And then changed my device record to be iotedge enabled: 
```
az iot hub device-identity update -d new_device_id -n MyHub --set 
capabilities.iotEdge=true
```

It _seems_ like everything is working like it should.... but I wasn't sure if doing this breaks everything in the back-end or something. 

Thanks in advance for the help,
Duncan

**Comment by myagley** [comment]
Unfortunately, the daemon hasn't been updated to do the DPS flow with symmetric keys. It currently only supports the TPM flow for DPS. There's currently no way to tell it to provision with DPS using a symmetric key.

This is an item on our backlog, but we won't be able to get to it until sometime early next year.

**Comment by duncanfinney** [comment]
Ahh that makes sense. We'll keep my eye out for the update! :+1: 

Is there any documentation around the differences between an IoT Hub provisioned device vs an IoT Edge provisioned device? I'm having trouble figuring out the line where IoT Hub ends and IoT Edge starts beyond that `capabilities.iotEdge` flag.

**Comment by myagley** [comment]
This has been released as part of the 1.0.7 release candidate. You can get packages here: https://github.com/Azure/azure-iotedge/releases/tag/1.0.7-rc2. This release will be GA in a few weeks.

**Comment by xwdreamer** [comment]
> This has been released as part of the 1.0.7 release candidate. You can get packages here: https://github.com/Azure/azure-iotedge/releases/tag/1.0.7-rc2. This release will be GA in a few weeks.

hi @myagley , my azure iot edge version is 1.0.7.1-1 

I found that  azure iot edge also didn't support  Symetric Key Attestation.
when  I use Symetric Key Attestation, I can't start iot edge.
I also ask this question in this issue: https://github.com/Azure/iotedge/issues/55

`
$ sudo apt-get install libiothsm iotedge
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'libiothsm-std' instead of 'libiothsm'
iotedge is already the newest version (1.0.7.1-1).
libiothsm-std is already the newest version (1.0.7.1-1).
The following packages were automatically installed and are no longer required:
  linux-headers-4.10.0-28 linux-headers-4.10.0-28-generic linux-headers-4.13.0-39 linux-headers-4.13.0-39-generic linux-headers-4.13.0-41 linux-headers-4.13.0-41-generic
  linux-headers-4.13.0-43 linux-headers-4.13.0-43-generic linux-headers-4.13.0-45 linux-headers-4.13.0-45-generic linux-headers-4.15.0-34 linux-headers-4.15.0-34-generic
  linux-headers-4.15.0-36 linux-headers-4.15.0-36-generic linux-headers-4.15.0-47 linux-headers-4.15.0-47-generic linux-headers-4.15.0-50 linux-headers-4.15.0-50-generic
  linux-image-4.10.0-28-generic linux-image-4.13.0-39-generic linux-image-4.13.0-41-generic linux-image-4.13.0-43-generic linux-image-4.13.0-45-generic
  linux-image-4.15.0-34-generic linux-image-4.15.0-36-generic linux-image-4.15.0-47-generic linux-image-4.15.0-50-generic linux-image-extra-4.10.0-28-generic
  linux-image-extra-4.13.0-39-generic linux-image-extra-4.13.0-41-generic linux-image-extra-4.13.0-43-generic linux-image-extra-4.13.0-45-generic linux-modules-4.15.0-34-generic
  linux-modules-4.15.0-36-generic linux-modules-4.15.0-47-generic linux-modules-4.15.0-50-generic linux-modules-extra-4.15.0-47-generic linux-modules-extra-4.15.0-50-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 215 not upgraded.
`
